net-misc/asterisk-g729/asterisk-g729-13.0.3.1.9.ebuild

```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit multilib
+
+DESCRIPTION="G.729 codec and supporting files for asterisk"
+HOMEPAGE="https://www.sangoma.com/asterisk/software/g729-codec/"
+
+AST_PV="$(ver_cut 1-2)"
+MY_PV="$(ver_rs 2 _)"
+
+SRC_URI="x86? (
+	https://downloads.digium.com/pub/telephony/codec_g729/asterisk-${AST_PV}/x86-32/codec_g729a-${MY_PV}-x86_32.tar.gz
+	https://downloads.digium.com/pub/register/x86-32/register -> astregister-x86_32
+	https://downloads.digium.com/pub/register/x86-32/asthostid -> asthostid-x86_32
+)
+amd64? (
+	https://downloads.digium.com/pub/telephony/codec_g729/asterisk-${AST_PV}/x86-64/codec_g729a-${MY_PV}-x86_64.tar.gz
+	https://downloads.digium.com/pub/register/x86-64/register -> astregister-x86_64
+	https://downloads.digium.com/pub/register/x86-64/asthostid -> asthostid-x86_64
+)"
+
+LICENSE="Digium"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+IUSE=""
+
+DEPEND=""
+RDEPEND="${DEPEND} =net-misc/asterisk-$(ver_cut 1)*"
+
+RESTRICT="mirror strip"
+
+QA_FLAGS_IGNORED_amd64="usr/lib64/asterisk/modules/codec_g729a.so"
+QA_FLAGS_IGNORED_x86="usr/lib/asterisk/modules/codec_g729a.so"
+QA_PREBUILT="usr/sbin/asthostid usr/sbin/astregister"
```

Comment:
I wasn't even aware that this _prefix is available for `QA_FLAGS_IGNORED`. However it seems to break `PKG_PREBUILT`. You'll have to use `$(get_libdir)` here, which in turns means you have to move this to `pkg_setup()`. You know, you could test whether this works or not with https://wiki.gentoo.org/wiki/Package_testing#make.conf these settings.

Reply:
Tony (I suspect) did those, way back from CVS days.

Ok, so reading the manpage, QA_PREBUILT is appended onto a number of other QA_ flags, including QA_FLAGS_IGNORED (I'm unable to locate the code that does this appending).

Assuming that the QA process will use QA_FLAGS_IGNORED_${ARCH} if it exists, and fall back to QA_FLAGS_IGNORED I can see how that won't work properly.  If it however checks both (which in my opinion would be the sane thing to do) this should work correctly.

Based on the code:

  7     local qa_var="QA_FLAGS_IGNORED_${ARCH/-/_}"
  8     eval "[[ -n \${!qa_var} ]] && QA_FLAGS_IGNORED=(\"\${${qa_var}[@]}\")"

(from /usr/lib/portage/python{2.7,3.6}/install-qa-check.d/10ignored-flags)

This does indeed look like an override.  Personally I would make that = on line 8 a += so that _${ARCH} gets appended instead of discarded.  Your opinion.

I do, however, not see any QA warnings, with or without the QA_PREBUILT line at all, the only way I can provoke a warning at all is with commenting the QA_FLAGS_IGNORED_* variables.

I've already had most of that make.conf sample exported into the environment before.

I've just dropped the PREBUILT line completely as it seems to have no effect.  If you're getting a QA warning, please do paste the warning here, I'm not getting one that I can find.

Reply:
```
 * Messages for package net-misc/asterisk-g729-13.0.3.1.9:

 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/sbin/asthostid
 * /usr/sbin/astregister
```

Reply:
*sigh* ... would you mind emailing me the applicable emerge --info for that?  Something is obviously different between our environments.